cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/expressions/BinaryOperator.kt

```diff
@@ -0,0 +1,203 @@
+            registerTypeListener((rhs as HasType.TypeListener))
+        }
+    }
+
+    private fun disconnectOldRhs() {
+        rhs.unregisterTypeListener(this)
+        if ("=" == operatorCode && rhs is HasType.TypeListener) {
+            unregisterTypeListener((rhs as HasType.TypeListener?)!!)
+        }
+    }
+
+    override fun typeChanged(src: HasType, root: MutableList<HasType>, oldType: Type) {
+        if (!TypeManager.isTypeSystemActive()) {
+            return
+        }
+        val previous = type
+        if (operatorCode == "=") {
+            setType(src.propagationType, root)
+        } else if (
+            "java.lang.String" == lhs.type.toString() || "java.lang.String" == rhs.type.toString()
+        ) {
+            // String + any other type results in a String
+            _possibleSubTypes.clear()
+            setType(TypeParser.createFrom("java.lang.String", language), root)
+        } else if (operatorCode == ".*" || operatorCode == "->*" && src === rhs) {
+            // Propagate the function pointer type to the expression itself. This helps us later in
+            // the call resolver, when trying to determine, whether this is a regular call or a function
+            // pointer call.
+            setType(src.propagationType, root)
+        }
+        if (previous != type) {
+            type.typeOrigin = Type.Origin.DATAFLOW
+        }
+    }
+
+    override fun possibleSubTypesChanged(src: HasType, root: MutableList<HasType>) {
+        if (!TypeManager.isTypeSystemActive()) {
+            return
+        }
+        val subTypes: MutableList<Type> = ArrayList(possibleSubTypes)
+        subTypes.addAll(src.possibleSubTypes)
+        setPossibleSubTypes(subTypes, root)
+    }
+
+    override fun toString(): String {
+        return ToStringBuilder(this, TO_STRING_STYLE)
+            .append("lhs", lhs.name)
+            .append("rhs", rhs.name)
+            .append("operatorCode", operatorCode)
+            .toString()
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
+        if (other !is BinaryOperator) {
+            return false
+        }
+        return super.equals(other) &&
+                lhs == other.lhs &&
+                rhs == other.rhs &&
+                operatorCode == other.operatorCode
+    }
+
+    override fun hashCode() = Objects.hash(super.hashCode(), lhs, rhs, operatorCode)
+
+    // We only want to supply a target if this is an assignment
+    override val target: AssignmentTarget?
+        get() = // We only want to supply a target if this is an assignment
+        if (isAssignment) (if (lhs is AssignmentTarget) lhs as AssignmentTarget? else null)
+            else null
+    override val value: Expression?
+        get() = if (isAssignment) rhs else null
+
+    private val isAssignment: Boolean
+        get() {
+            // TODO(oxisto): We need to discuss, if the other operators are also assignments and if
```

Comment:
Just stumbled over this, I think the other operators are assignments too, however, the full list of assignments should be retrieved from the language frontend or trait

Reply:
Yes, I think this is partly addressed in https://github.com/Fraunhofer-AISEC/cpg/pull/1025